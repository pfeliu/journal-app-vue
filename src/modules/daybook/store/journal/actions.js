// export const myAction = async({ commit }) => {

import journalApi from '@/api/journalApi';

// }

export const loadEntries = async ({ commit }) => {
  const { data } = await journalApi.get('/entries.json');
  // console.log(data);

  if (!data) {
    commit('setEntries', []);
    return;
  }
  const entries = [];
  for (let id of Object.keys(data)) {
    entries.push({ id, ...data[id] });
  }

  console.log(entries);
  commit('setEntries', entries);
};
export const updateEntry = async ({ commit }, entry) => {
  const { date, picture, text } = entry;
  const dataToSave = { date, picture, text };

  await journalApi.put(`/entries/${entry.id}.json`, dataToSave);

  commit('updateEntry', { ...entry });
};
export const createEntry = async ({ commit }, entry) => {
  console.log(entry);
  const { date, picture, text } = entry;
  const dataToSave = { date, picture, text };

  const { data } = await journalApi.post(`/entries.json`, dataToSave);

  dataToSave.id = data.name;

  commit('addEntry', dataToSave);

  return data.name;
};
export const deleteEntry = async ({ commit }, id) => {
  await journalApi.delete(`/entries/${id}.json`);

  commit('deleteEntry', id);

  return null;
};
