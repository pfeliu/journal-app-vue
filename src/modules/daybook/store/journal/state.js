export default () => ({
  isLoading: true,
  entries: [
    {
      id: new Date().getTime(),
      date: new Date().toDateString(),
      text: 'Aliqua duis magna aliqua reprehenderit aliquip ipsum minim deserunt eiusmod aliqua ex aliqua. Exercitation id labore laborum ex tempor sint dolor incididunt commodo fugiat deserunt consequat sunt proident. Dolor velit tempor et aliqua pariatur.',
      picture: null,
    },
    {
      id: new Date().getTime() + 1000,
      date: new Date().toDateString(),
      text: 'Aliqua duis magna aliqua reprehenderit aliquip ipsum minim deserunt eiusmod aliqua ex aliqua. Exercitation id labore laborum ex tempor sint dolor incididunt commodo fugiat deserunt consequat sunt proident. Dolor velit tempor et aliqua pariatur.',
      picture: null,
    },
    {
      id: new Date().getTime() + 2000,
      date: new Date().toDateString(),
      text: 'Aliqua duis magna aliqua reprehenderit aliquip ipsum minim deserunt eiusmod aliqua ex aliqua. Exercitation id labore laborum ex tempor sint dolor incididunt commodo fugiat deserunt consequat sunt proident. Dolor velit tempor et aliqua pariatur.',
      picture: null,
    },
  ],
});
