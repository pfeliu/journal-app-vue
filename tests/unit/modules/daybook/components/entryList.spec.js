import { createStore } from 'vuex';
// import { getEntriesByTerm } from '@/modules/daybook/store/journal/getters';
import { journalState } from '../../../mock-data/test-journal-state';
import { shallowMount } from '@vue/test-utils';
import EntryList from '@/modules/daybook/components/EntryList.vue';
import journal from '@/modules/daybook/store/journal';

const createVuexStore = (initialState) =>
  createStore({
    modules: {
      journal: {
        ...journal,
        state: { ...initialState },
      },
    },
  });

describe('Entry List', () => {
  /* const journalMockModule = {
    namespaced: true,
    getters: {
      // getEntriesByTerm: jest.fn(),
      getEntriesByTerm,
    },
    state: () => ({
      isLoading: false,
      entries: journalState.entries,
    }),
  };

  const store = createStore({
    modules: {
      journal: { ...journalMockModule },
    },
  }); */

  const store = createVuexStore(journalState);

  const mockRouter = {
    push: jest.fn(),
  };

  let wrapper;
  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallowMount(EntryList, {
      global: {
        mocks: {
          $router: mockRouter,
        },
        plugins: [store],
      },
    });
  });

  test('should call entries by term', () => {
    expect(wrapper.findAll('entry-stub').length).toBe(2);
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should call getEntriesByTerm and filter', async () => {
    const input = wrapper.find('input');
    await input.setValue('segunda');

    expect(wrapper.findAll('entry-stub').length).toBe(1);
  });

  test('should bottom redirect to root', () => {
    wrapper.find('button').trigger('click');

    expect(mockRouter.push).toHaveBeenCalledWith({
      name: 'entry',
      params: { id: 'new' },
    });
  });
});
