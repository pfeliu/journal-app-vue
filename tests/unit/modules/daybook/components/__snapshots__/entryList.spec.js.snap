// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry List should call entries by term 1`] = `
<div class="entry-list-container">
  <div class="px-2 pt-2"><input type="text" class="form-control" placeholder="Search entry"></div>
  <div class="mt-2 d-flex flex-column"><button class="btn btn-primary mx-3"><i class="fa fa-plus-circle"></i> New Entry </button></div>
  <div class="entry-scrollarea">
    <entry-stub entry="[object Object]"></entry-stub>
    <entry-stub entry="[object Object]"></entry-stub>
  </div>
</div>
`;
